File: 6-zigzag-conversion/6-zigzag-conversion.java
class Solution {
    public String convert(String s, int numRows) 
    {
        if(numRows==1)
            return s;
        String res = "";
        String[][] arr = new String[numRows][s.length()];
        int i=0;
        int j=0;
        while(j!=s.length())
        {
            for(int x=0; x<numRows && j<s.length(); x++)
            {
                arr[x][i]=String.valueOf(s.charAt(j));
                // System.out.println(x+" "+i+" "+arr[x][i] +" "+j);
                j++;
            }
            i++;
            for(int y=numRows-2; y>0 && j<s.length(); y--)
            {
                arr[y][i]=String.valueOf(s.charAt(j));
                // System.out.println(y+" "+i+" "+arr[y][i]+" "+ j);
                j++;
            }
            i++;
        }
        for(i=0; i<arr.length; i++){
            for(j=0; j<arr[0].length; j++){
                if(arr[i][j]!=null){
                    res+=arr[i][j];
                }
            }
        }
        
        // for(int a=0; a<arr.length; a++)
        // {
        //     for(int b=0; b<arr[0].length; b++)
        //     {
        //         System.out.print(arr[a][b]+" ");
        //     }
        //     System.out.println();
        // }
        
        return res;
    }
}

File: 6-zigzag-conversion/6-zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if(numRows == 1) return s;
        int n = s.size();
        char v[numRows][n];
        int rows = sizeof(v)/sizeof(v[0]);
        int cols = sizeof(v[0])/sizeof(v[0][0]);
        for(int i=0; i<rows; i++){
            for(int j=0; j<cols; j++){
                // if(v[i][j]>='A' && v[i][j]<='Z'){
                //     res+=v[i][j];
                // }
                v[i][j] = ' ';
            }
        }
        int j = 0;
        string res="";
        int i=0;
        while(j<n){
            for(int x=0; x<numRows && j<n; x++)
            {
                v[x][i]=s[j];
                // cout<<x<<" "<<i<<" "<<j<<" "<<v[x][i]<<" "<<endl;   
                j++;
            }
            i++;
            for(int y=numRows-2; y>0 && j<n; y--)
            {
                v[y][i]=s[j];
                 // cout<<y<<" "<<i<<" "<<j<<" "<<v[y][i]<<" "<<endl;
                j++;
                i++;
            }
            // i++;
        }
        
        // cout<<rows<<cols;
        for(i=0; i<rows; i++){
            for(j=0; j<cols; j++){
                // if((v[i][j]>='A' && v[i][j]<='Z') || ((v[i][j]>='a' && v[i][j]<='z'))){
                //     res+=v[i][j];
                //     cout<<v[i][j];
                // }
                if(v[i][j]!=' ')
                    res+=v[i][j];
                
            }
            // cout<<endl;
        }
        cout<<"res"<<res<<endl;
        return res;
    }
};